### ISSUE TITLE
How to send Chrome DevTools Protocol message from a custom DevTools panel

### ISSUE THREAD

**Issue by NickolasBoyer** [ChromeDevTools/devtools-protocol#187]
I think I'm overlooking something very obvious but I can't seem to figure this out:

How does one send messages from a Chrome extension (with a `devtools_page` so it's running in the DevTools context as an extra panel in the DevTools window)

Do I need to connect directly to a websocket server? Currently, the only solution I can seem to find is running Chrome with `--remote-debugging-port=9222` and connecting to that WS server but since the entire DevTools window itself is basically a webapp that is already doing this without the remote port I assume this is not necessary?

Basically I have no idea and any help would be greatly appreciated, thanks!

**Comment by NickolasBoyer** [comment]
It indeed ended up being quite obvious but hard to find. In case someone else ever runs into this: You can pass messages through the `chrome.debugger.sendCommand` method, you'll need to add the debugging permission to your extension by specifying it in the `manifest.json` file. Also don't forget to `attach` the debugger before trying to use `sendCommand`. Happy coding, [the docs are here](https://developer.chrome.com/extensions/debugger#method-sendCommand), I'm closing this issue.

**Comment by NickolasBoyer** [comment]
[the docs are here](url)
